navit/xslt/osd_minimum.xslt

```diff
@@ -5,27 +5,35 @@
       <xsl:param name="NEXT_TURN_TEXT_HIGHT"><xsl:value-of select="round(20*number($OSD_SIZE))"/></xsl:param>
       <xsl:param name="OSD_USE_OVERLAY"><xsl:value-of select="$OSD_USE_OVERLAY='yes' or $OSD_USE_OVERLAY='true' or $OSD_USE_OVERLAY='1'"/></xsl:param>
 
-      <osd type="compass" enabled="yes" x="{round(-60*number($OSD_SIZE))}" y="{round(-80*number($OSD_SIZE))}" w="{round(60*number($OSD_SIZE))}" h="{round(80*number($OSD_SIZE))}" font_size="{round(200*number($OSD_SIZE))}" enable_expression="vehicle.position_valid"/>
+      <osd type="text" label="${{navigation.item[1].name}}" x="0" y="0" w="100%" h="{round(14*number($OSD_SIZE))}" font_size="{round(200*number($OSD_SIZE))}" enable_expression="navigation.nav_status&gt;=3"/>
       <xsl:text>&#x0A;        </xsl:text>
-      <osd type="text" label="${{navigation.item.destination_length[named]}}\n${{navigation.item.destination_time[arrival]}}" x="{round(-60*number($OSD_SIZE))}" y="0" w="{round(60*number($OSD_SIZE))}" h="{round(40*number($OSD_SIZE))}" font_size="{round(200*number($OSD_SIZE))}" enable_expression="navigation.nav_status>=3"/>
+      <osd type="compass" enabled="yes" x="{round(-60*number($OSD_SIZE))}" y="{round(-94*number($OSD_SIZE))}" w="{round(60*number($OSD_SIZE))}" h="{round(80*number($OSD_SIZE))}" font_size="{round(200*number($OSD_SIZE))}" enable_expression="vehicle.position_valid"/>
```

Comment:
Again, is there a better way. Need to think how to convert this to .svg for Sailfish using xslt. At least it features "w" and "h", which is mandatory for pure .svg

Reply:
Probably not something that will work across all platforms, as not every platform supports SVG. The best bet is probably to use XSLT to convert this to SVG on platforms which support it. Question—will SVGs be scaled to match the size specified? PNGs appear to get cropped or padded, at least for `toggle_announcer`.

Reply:
About file types: Doesn't the graphics subsystem try to "guess" the filename anyway? I remember something in navit/graphics.c (image_new_helper). So it should be ok to give "w" and "h" properly and then remove the file extension and the size from the file name. Graphics subsystem will pick what's there, starting with %name_%width_%height.png Next try would be %name.svg and so fourth. That's what I think we should use for everybody to be fine.

About scaling: Unfortunately this is up to the graphics plugin to handle correctly. And obviously not all do right. Qt5 does. If you see that cropping, then the graphics used by you does not. But bitmap scaling should only be the last resort. If using bitmaps, then they should be pre scaled as you do as well.

About svg scaling: Svg's as all other file formats do include a size hint, So if no size is explicitly given, at least QT5 uses the size of their size hint to render them to bitmap. Unfortunately this size hint is often bogus, as nobody cares for that on svg usually. So one has to provide the size to the renderer explicitly. That's why using "w" and "h" properties is so important and should be preferred over implicitly setting the size by the bitmap icon size. The one svg file can be rendered to all required sizes on the fly.